tracer/src/Datadog.Trace/Agent/MessagePack/SpanMessagePackFormatter.cs

```diff
@@ -275,14 +298,30 @@ private int WriteMetrics(ref byte[] bytes, int offset, in SpanModel model, ITagP
             offset = tagWriter.Offset;
             count += tagWriter.Count;
 
-            if (model.IsLocalRoot)
+            // add "process_id" tag to local root span (if present)
+            if (model.IsLocalRoot && _processIdValueBytes != null)
             {
-                // add "process_id" tag
-                if (_processIdValueBytes != null)
+                count++;
+                offset += MessagePackBinary.WriteStringBytes(ref bytes, offset, _processIdNameBytes);
+                offset += MessagePackBinary.WriteRaw(ref bytes, offset, _processIdValueBytes);
+            }
+
+            // add "_sampling_priority_v1" tag to all "chunk orphans"
+            // (spans whose parents are not found in the same chunk)
+            if (model.IsChunkOrphan && model.TraceChunk.SamplingPriority is { } samplingPriority)
+            {
+                count++;
+                offset += MessagePackBinary.WriteStringBytes(ref bytes, offset, _samplingPriorityNameBytes);
+
+                if (samplingPriority is >= -1 and <= 2)
+                {
+                    // values begin at -1, so they are shifted by 1 from their array index: [-1, 0, 1, 2]
+                    offset += MessagePackBinary.WriteRaw(ref bytes, offset, _samplingPriorityValueBytes[samplingPriority + 1]);
+                }
+                else
                 {
-                    count++;
-                    offset += MessagePackBinary.WriteStringBytes(ref bytes, offset, _processIdNameBytes);
-                    offset += MessagePackBinary.WriteRaw(ref bytes, offset, _processIdValueBytes);
+                    // fallback to support unknown future values that are not cached
+                    offset += MessagePackBinary.WriteDouble(ref bytes, offset, samplingPriority);
```

Comment:
NitPicking - Should we extend the `_samplingPriorityValueBytes` array to add the unknown value instead? If we receive a new value, there's a good chance we receive it multiple times

Reply:
Note that we use the sampling priority value to index into the array:
```csharp
var bytes = _samplingPriorityValueBytes[samplingPriority + 1]);
```

Extending the array dynamically makes sense if the new value is `3`, for example, because the array only grows in size by 1. But if the new value is `-2`, we would have to change how we use the array because now the offset above needs to be  2 instead of 1.

And if the new value is `10_000` we could need an array of size 10_002.
```
[-1, 0, 1, 2, ... 10_000]
```

It's not impossible, but I don't think it's worth it. I doubt we can add a new sampling priority for a long time, anyway. We tried that recently, but many traces were validating the value (they would not allow `< -1` or `> 2`). And that's how sampling mechanism was born!